### PR TITLE
CDI-471 Introduce SPI for retrieving multiple annotations of the same…

### PR DESCRIPTION
**Pull request by johnament**
… type, deprecating the old one.  Updated spec docs for qualifiers to reflect repeatability and spec docs for annotated for new method.

### CODE REVIEW

**Comment by johnament** [comment]
One thing strikes me about this - is it possible people implemented their own annotated?  I can't think of a reason right now.

File: spec/src/main/asciidoc/core/definition.asciidoc [hunk 1]
```diff
@@ -382,6 +382,67 @@ PaymentProcessor getPaymentProcessor(@Synchronous PaymentProcessor sync,
 }
 ----
 
+[[repeating_qualifiers]]
```
Comment (tremes) [inline spec/src/main/asciidoc/core/definition.asciidoc]: Great.:-) Appreciate.

File: api/src/main/java/javax/enterprise/inject/spi/Annotated.java [hunk 2]
```diff
@@ -51,18 +52,34 @@
...
      * @return a set of all types to which the base type should be considered assignable
      */
-    public Set<Type> getTypeClosure();
+    Set<Type> getTypeClosure();
 
     /**
      * <p>
      * Get program element annotation of a certain annotation type.
+     *
+     * Since 2.0, calling this method is considered deprecated.  To support {@see Repeatable} annotations, call {@see getAnnotation}
```
Comment (mkouba) [inline api/src/main/java/javax/enterprise/inject/spi/Annotated.java]: I would use the "settled" wording (see for example `BeanManager.createInjectionTarget()`):
> This method is deprecated from CDI 2.0 and {@link #getAnnotations(Class)} should be used instead.

Also I don't think it's possible to use `{@see}` inline.

File: api/src/main/java/javax/enterprise/inject/spi/Annotated.java [hunk 3]
```diff
@@ -51,18 +52,34 @@
...
      * </p>
      * 
      * @param <T> the type of the annotation
      * @param annotationType the class of the annotation type
      * @return the program element annotation of the given annotation type, or a null value
```
Comment (mkouba) [inline api/src/main/java/javax/enterprise/inject/spi/Annotated.java]: _the **first**  program element annotation found, or a null value_?

File: api/src/main/java/javax/enterprise/inject/spi/Annotated.java [hunk 4]
```diff
@@ -51,18 +52,34 @@
...
      */
-    public <T extends Annotation> T getAnnotation(Class<T> annotationType);
+    @Deprecated
```
Comment (mkouba) [inline api/src/main/java/javax/enterprise/inject/spi/Annotated.java]: AFAIK we never use `@Deprecated` in CDI API.

File: api/src/main/java/javax/enterprise/inject/spi/Annotated.java (hunk 2 above)
Comment (johnament) [inline api/src/main/java/javax/enterprise/inject/spi/Annotated.java]: Ok, I'll replace the `@Deprecated` with this as well.  Link was what I was looking for, but couldn't remember.

**Comment by johnament** [comment]
@mkouba all fixed, I hope!

**Comment by mkouba** [comment]
@johnament Looks good!

**Pull request merged**